druid/src/window.rs

```diff
@@ -250,24 +250,33 @@ impl<T: Data> Window<T> {
         env: &Env,
         process_commands: bool,
     ) {
-        let mut base_state = BaseState::new(self.root.id());
-        let mut ctx = LifeCycleCtx {
-            command_queue: queue,
-            window_id: self.id,
-            base_state: &mut base_state,
-        };
-
         if let LifeCycle::AnimFrame(_) = event {
-            self.do_anim_frame(&mut ctx, data, env)
+            self.do_anim_frame(queue, data, env)
         } else {
+            let mut base_state = BaseState::new(self.root.id());
+            let mut ctx = LifeCycleCtx {
```

Comment:
hmm, I think I'm missing the reason we can't reuse the context here?

Reply:
It's the borrow checker. `LifeCycleCtx` holds an immutable reference to `self` while `do_anim_frame` wants a mutable reference to `self`.